python-avd/pyavd/_eos_designs/structured_config/connected_endpoints/utils.py

```diff
@@ -72,7 +72,9 @@ def _filtered_network_ports(self: AvdStructuredConfigConnectedEndpoints) -> list
         for index, network_port in enumerate(self.inputs.network_ports):
             network_port._context = f"network_ports[{index}]"
             network_port_settings = self.shared_utils.get_merged_adapter_settings(network_port)
-            if not self._match_regexes(network_port_settings.switches, self.shared_utils.hostname):
+            if network_port_settings.switches and not self._match_regexes(network_port_settings.switches, self.shared_utils.hostname):
```

Comment:
this change is changing the behavior. In the current implementation, if `switches` is not set, then no network_port are configured on any device. The change, if neither `switches`, nor `platforms` are set would configure the network_ports on ALL devices.

you would need an extra `if` to make sure that either `switches` or `platforms` are set + add a test case with network_ports without `switches` not `platforms` set to validate the behavior

Reply:
Fixed behavior to ensure match criteria is required, added test to ensure configuration is not generated if missing both switch hostname and platform match criteria.